src/lib/coda_base/pending_coinbase.ml

```diff
@@ -0,0 +1,552 @@
+open Core_kernel
+open Import
+open Snarky
+open Snark_params
+open Snark_params.Tick
+open Let_syntax
+open Currency
+open Fold_lib
+open Snark_bits
+
+let coinbase_tree_depth = Snark_params.pending_coinbase_depth
+
+(*Total number of stacks*)
+let coinbase_stacks = Int.pow 2 coinbase_tree_depth
+
+module Coinbase_data = struct
+  type t = Public_key.Compressed.t * Amount.Signed.t [@@deriving sexp]
```

Comment:
Reminder to myself to create an issue to reuse the hash section of the proposer public key.